server/entity/effect/effect.go

```diff
@@ -57,6 +57,21 @@ func NewAmbient(t LastingType, lvl int, d time.Duration) Effect {
 	return Effect{t: t, lvl: lvl, d: d, ambient: true}
 }
 
+// potentType represents an effect type which can have its potency changed.
+type potentType interface {
+	Type
+	// WithPotency updates the potency of the type with the one given and returns it.
+	WithPotency(potency float64) Type
+}
+
+// WithPotency returns a new effect with the given potency, if the effect type allows for it.
+func (e Effect) WithPotency(potency float64) Effect {
```

Comment:
I don't think we should expose this as a method. I think it'd be a better idea if we exposed `PotentType`. This method doesn't do anything for most effects.